juju/client/proxy/kubernetes/proxy.py

```diff
@@ -60,6 +60,7 @@ def connect(self):
 
     def __del__(self):
         self.close()
+        os.unlink(self.temp_ca_file.name)
```

Comment:
Let's handle the `FileNotFoundError` here ourselves and log a nice warning, instead of `__del__` ignoring it and writes it onto the `stderr` verbatim. (exceptions raised within destructors are suppressed -- to avoid weird behaviors, like the exception handler trying to destroy the object -- thereby calling `__del__` again -- [see doc](https://docs.python.org/3/reference/datamodel.html#object.__del__))

We really should stop using things like `__del__` and turn this into a context by adding `__enter__` and `__exit__` to handle these things, but that's beyond the scope of this PR.